### PR TITLE
Replaced gmtime with gmtime_r

### PR DESCRIPTION
**Pull request by olehermanse**
gmtime_r is thread safe.

Reported by LGTM:
https://lgtm.com/rules/2154840805/

Signed-off-by: Ole Herman Schumacher Elgesem <ole.elgesem@northern.tech>

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom please run jenkins build, I wish you could do it with exotics ;)

**Comment by olehermanse** [comment]
@vpodzime please review.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1036/)

https://ci.cfengine.com/job/pr-pipeline/1036/

**Comment by olehermanse** [comment]
@cf-bottom jenkins.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1038/)

https://ci.cfengine.com/job/pr-pipeline/1038/

File: cf-upgrade/log.c [hunk 1]
```diff
@@ -47,11 +48,14 @@ static char *prepare_message(char *format, va_list args)
     int message_size = 0;
     char timestamp[] = "YYYY/MM/DD HH:MM:SS";
     char buffer[MAX_LOG_ENTRY_SIZE];
-    time_t now = time(NULL);
-    struct tm *now_tm = gmtime(&now);
+    struct tm now;
+    {
+        time_t now_seconds = time(NULL);
+        gmtime_r(&now_seconds, &now);
+    }
```
Comment (vpodzime) [inline cf-upgrade/log.c]: This nested block is really confusing here. It looks to me like if the `struct` was initialized using itself at the first glance. Please add a blank line between the declaration and the block or get rid of the nested block.

**Comment by olehermanse** [comment]
Started another build:

[Build Status](https://ci.cfengine.com/job/pr-pipeline/1051/)

**Comment by olehermanse** [comment]
@vpodzime Fixed!

**Comment by olehermanse** [comment]
Errors look unrelated, merging.

**Pull request merged**